### ISSUE TITLE
"a container has already joined the endpoint" when no container running

### ISSUE THREAD

**Issue by alvinr** [docker/libnetwork#369]
# Problem
Occasionally get these messages

```Cannot start container b255f6a838acbd34ea93f76dfe28a6b629d06a9323496f3fa981e2f93a01781b: a container has already joined the endpoint```

There are currently no running containers (other than the swarm containers). Looking at Consul I see a single value

```
$ curl 172.16.145.138:8500/v1/kv/?keys
["docker/libnetwork/endpoint/bb53beeebab193411ff85d6c4e46208ff054333b29fef5b11c4906c471ffd65a/93c54085fb3770d2e7b1406fe6fdb912619d5432839d7afd87fbff9b63045f48/","docker/libnetwork/endpoint/bb53beeebab193411ff85d6c4e46208ff054333b29fef5b11c4906c471ffd65a/b3210bbe9546074f9de973de540e15bf324e47b43dd3beb77a999b2d2180c328/","docker/libnetwork/endpoint/bb53beeebab193411ff85d6c4e46208ff054333b29fef5b11c4906c471ffd65a/cba678b15488ec8b07dd149ff4ee0dd4af8296b26e40ca76eb1061712e03309a/","docker/libnetwork/idm/ipam-id/","docker/libnetwork/idm/vxlan-id/","docker/libnetwork/network/bb53beeebab193411ff85d6c4e46208ff054333b29fef5b11c4906c471ffd65a/","docker/libnetwork/overlay/network/bb53beeebab193411ff85d6c4e46208ff054333b29fef5b11c4906c471ffd65a/"]
```

# Reproduce
The configuration is a 5 node Swarm cluster, where I am going through the following cycle repeatedly. At some point I will get this message

* start 10-20 containers (via Compose)
* docker service detach prod_mongodb_1 prod_mongodb_1.multihost (on swarm-0)
* docker service attach mongodb_mongodb_1 prod_mongodb_1.multihost (on swarm-0)
* docker-compose stop
* docker-compose rm -f

# Environment

```
docker@swarm-0:~$ uname -a
Linux swarm-0 4.0.5-boot2docker #1 SMP Tue Jun 16 01:39:56 UTC 2015 x86_64 GNU/Linux
```

```
docker --version
Docker version 1.8.0-dev, build 91fd45c-dirty, experimental
```

```
docker info
Containers: 0
Images: 0
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 0
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.0.3-boot2docker
Operating System: Boot2Docker 1.6.2 (TCL 5.4); master : 4534e65 - Wed May 13 21:24:28 UTC 2015
CPUs: 1
Total Memory: 998 MiB
Name: dev
ID: LZBF:J6FU:45BS:N4BU:3TMO:QNMR:PUS7:LVB6:NADW:4KZJ:MJ7E:XLMX
Debug mode (server): true
File Descriptors: 9
Goroutines: 15
System Time: 2015-07-09T20:10:39.437985656Z
EventsListeners: 0
Init SHA1: 
Init Path: /usr/local/bin/docker
Docker Root Dir: /mnt/sda1/var/lib/docker
Labels:
 provider=vmwarefusion
Experimental: true
```

**Comment by alvinr** [comment]
# Workaround
if you restart the docker daemons it clears this problem and you can restart the container.

**Comment by dave-tucker** [comment]
Closing as this should have been fixed in #440
